www/front_src/src/Extensions/Listing/index.tsx

```diff
@@ -127,12 +130,46 @@ const ExtensionsManager = ({ reloadNavigation }): JSX.Element => {
     }).then(({ status, result }) => {
       if (status) {
         setExtension(result as Extensions);
+        console.log(result);
       } else {
         showErrorMessage(result as string);
       }
     });
   }, [getAppliedFilterCriteriasAtom]);
 
+  React.useEffect(() => {
+    const notInstallableExtensionModuleExiste = !isEmpty(
+      filter(
+        pathEq(['version', 'installed'], false),
+        extensions.module.entities,
+      ),
+    );
+
+    const notInstallableExtensionWidgetExiste = !isEmpty(
+      filter(
+        pathEq(['version', 'installed'], false),
+        extensions.widget.entities,
+      ),
+    );
+
+    setInstallable(
+      notInstallableExtensionModuleExiste ||
+        notInstallableExtensionWidgetExiste,
+    );
+
+    const notUpdatableExtensionModuleExiste = !isEmpty(
+      filter(pathEq(['version', 'outdated'], true), extensions.module.entities),
+    );
+
+    const notUpdatableExtensionWidgetExiste = !isEmpty(
+      filter(pathEq(['version', 'outdated'], true), extensions.widget.entities),
+    );
+
+    setUpdatable(
+      notUpdatableExtensionModuleExiste || notUpdatableExtensionWidgetExiste,
+    );
+  }, [extensions]);
+
```

Comment:
I would'nt use states here but simply variable to optimize the render cycles

Reply:
can i leave useEffect ???